### ISSUE TITLE
React-RxJS release roadmap

### ISSUE THREAD

**Issue by josepot** [re-rxjs/react-rxjs#57]
These are the things that I think should happen before we publish the first stable release:

- [x] Split the library between `core`, `utils` and `dom`.
    - [x] `@react-rxjs/core` should have everything that `react-rxjs` already has except for the `utils` section
    - [x] `@react-rxjs/utils` I would open a separate issue to discuss what we should put in there, but in a nutshell it will have the APIs that are currently under the `utils` section plus some useful operators from `@josepot/rxjs-utils` (like `groupInMap`, `mergeWithKey`, `recursiveObservable`... We will see)
    - [x] `@react-rxjs/dom` will expose an operator named `batchUpdates` that won't be necessary with React 17. It's basically a `observeOn(asapScheduler)` that does the `next` on the observer wrapped on a ReactDom `unstable_batchedupdates`.

- [x] Docs site. I'm the owner of the domain `react-rxjs.org` and I would like to have a site very similar to the one that [recoil has](https://recoiljs.org/docs/introduction/installation). These are the sections that I would like to have in the initial docs site:

    - [x] Motivation: The main idea of this section would be to explain why it's so limiting and frustrating to use React as a state-management library, the main reason is that it's not Reactive. Then we would explain why it's so desirable to use a truly reactive state-management system with React: code-navigability, avoid indirection, automatic code-splitting, being able to leverage React-Suspense and React-Error-Boundaries and the coming features of React Concurrent Mode, while having top-notch performance because with a forward-referenced based architecture we would only trigger the strictly necessary re-renders.

    - [x] Core concepts: In this section I would explain that thanks to the great improvements that React has made during these last years, it's now possible to have a set of RxJS bindings for React that address the impedance mismatch that happens when trying to use RxJS directly in React. It does so by making streams first-class citizens through hooks that leverage React Suspense and React Error-Boundaries. It also provides a means of multicasting the streams that are being used by React in a way that avoids memory-leaks, that makes composition easier while also avoids premature unsubscriptions because a component gets mounted/unmoutned. In this section why the "shareLatest" operator is a key and core piece for coveting this impedance mismatch.

    - [x] Installation: It would be almost identical to the installation section of Recoil

    - [x] Getting Started: We have to think of an example that showcases the power of the library, that's easy to understand for RxJS newcomers and that's not too trivial... I don't think that we would want to use the same example that they use in the Getting Started section of Recoil. I think that example makes sense for their library, but not so much for ours... I'm still not sure.

    - [x] TODOs App: I mean, I guess that it's almost mandatory to show an example of how to build the freaking todos app... That's what everybody does, and most developers have seen it implemented in so many different ways that I guess that we should have it.

    - [x] Real-life example: Let's not be shy, let's use this https://github.com/josepot/rtk-github-issues-example/pull/1/files It's a great example to explain how this library integrates so nicely with React Suspense and ErrorBoundaries, while making code-splitting a walk in the park.

    - [x] API Reference: Move and improve the API reference that we have in the README of the Repo to the docs website.

cc: @voliva @bhavesh-desai-scratch @rikoe @egillh210 thoughts?

**Comment by rikoe** [comment]
@josepot I assume the first point would mean adding lerna?

I don't think all of these are necessary before the first stable release (e.g. the TODO app) - nice-to-haves certainly, but I think getting the library released is also important, and you could then independently iterate on some of the surrounding things that would not require a new version of the library.

**Comment by react-rxjs** [comment]
It's a great project

**Comment by josepot** [comment]
> It's a great project

@react-rxjs shame on you for squatting on that handle-name. FTR: You are not going to get a dime from us.

**Comment by react-rxjs** [comment]
@josepot What do you mean？I registered in 2017 and wanted to do something about RXJS, but I just didn't have the time. I just got an email about @, so I came to check it out.

**Comment by josepot** [comment]
@rikoe 

> @josepot I assume the first point would mean adding lerna?

I wasn't sure what the best way to proceed was, so yesterday night I asked @dylanvann for advice (he is a freaking mono-repo guru, for real!). So, after that conversation we realized that since our packages don't depend on each other, then it probably makes more sense to just use yarn-workspaces and [wsrun](https://www.npmjs.com/package/wsrun). He helped me to set it up and as you can see in #58 it seems to work quite nicely. So, at least for now, I think that we will stick to that.

> I don't think all of these are necessary before the first stable release (e.g. the TODO app) - nice-to-haves certainly, but I think getting the library released is also important, and you could then independently iterate on some of the surrounding things that would not require a new version of the library.

💯  Yep! This is a very good point. You made me change my mind, let's focus on a MVP release first... Even if it doesn't have all the examples/tutorials... We can add those later, for sure! Let me think a bit more on how that MVP should look like, but yep, let's get something out of the door ASAP.

**Comment by josepot** [comment]
FYI @rikoe https://github.com/re-rxjs/react-rxjs.org/pull/1

**Comment by CMCDragonkai** [comment]
Question, why did the utils get separated out? They seem necessary for usage of react-rxjs?
